Brokerages/Zerodha/ZerodhaBrokerage.HistoryProvider.cs

```diff
@@ -0,0 +1,277 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Brokerages.Zerodha
+{
+    /// <summary>
+    /// ZerodhaBrokerage - IHistoryProvider implementation
+    /// </summary>
+    public partial class ZerodhaBrokerage
```

Comment:
As mentioned in the other comment, I think we should just implement `IBrokerage.GetHistory()`, returning an `IEnumerable<BaseData>` instead of `IEnumerable<Slice>`